src/include/distributed/multi_client_executor.h

```diff
@@ -72,6 +75,28 @@ typedef enum
 } BatchQueryStatus;
 
 
+/* Enumeration to track what the task associated with a connection is waiting for */
+typedef enum WaitStatus
```

Comment:
I don't like ConnStatus much, several things aren't related to the connection itself. What do you think about
```
/* Enumeration to track whether a task is ready to run and, if not, what it's blocked on*/
typedef enum TaskExecutionStatus
{
	TASK_STATUS_INVALID = 0,
	TASK_STATUS_ERROR, /* error occured */
	TASK_STATUS_READY, /* task ready to be processed further */
	TASK_STATUS_SOCKET_READ, /* waiting for connection to become ready for reads */
	TASK_STATUS_SOCKET_WRITE /* waiting for connection to become ready for writes */
} TaskExecutionStatus;
```